skrub/datasets/_utils.py

```diff
@@ -54,3 +81,155 @@ def get_data_dir(name=None, data_home=None):
     if name is not None:
         data_dir = data_dir / name
     return data_dir
+
+
+def load_simple_dataset(dataset_name, data_home=None):
+    """TODO
+    """
+    bunch = _load_dataset_files(dataset_name, data_home)
+    bunch["X"] = bunch.pop(dataset_name)
+    if (target := bunch.metadata.get("target", None)) is not None:
+        bunch["y"] = bunch["X"][target]
+        bunch["X"] = bunch["X"].drop(columns=target)
+    return bunch
+
+
+def _load_dataset_files(dataset_name, data_home):
+    """
+    TODO
+    skrub_data/
+        fraud/
+            fraud.tar.gz
+            fraud/
+                baskets.csv
+                products.csv
+                metadata.json
+    """
+    data_home = get_data_home(data_home)
+    dataset_dir = data_home / dataset_name
+    datafiles_dir = dataset_dir / dataset_name
+    datafiles_dir.mkdir(parents=True, exist_ok=True)
+
+    archive_path = dataset_dir / f"{dataset_name}.zip"
+    metadata = DATASET_INFO[dataset_name]
+
+    diff_checksum = False
+    if archive_path.exists():
+        expected_checksum = metadata["sha256"]
+        checksum = _sha256(archive_path)
+        if (diff_checksum := expected_checksum != checksum):
+            warnings.warn(
+                f"SHA256 checksum of existing local file {archive_path.name} "
+                f"({checksum}) differs from expected ({expected_checksum}): "
+                f"re-downloading from {metadata['urls']} ."
+            ) 
+
+    if not any(datafiles_dir.iterdir()) or diff_checksum:
+        _extract_archive(dataset_dir, archive_path, diff_checksum)
+
+    bunch = Bunch()
+    for file_path in datafiles_dir.iterdir():
+        if file_path.suffix == ".csv":
+            bunch[file_path.stem] = pd.read_csv(file_path)
+        elif file_path.suffix == ".json":
+            bunch[file_path.stem] = json.loads(file_path.read_text(encoding="utf-8"))
+
+    return bunch
+
+
+def _extract_archive(dataset_dir, archive_path, diff_checksum):
+    dataset_name = dataset_dir.name
+
+    if not archive_path.exists() or diff_checksum:
+        _download_archive(dataset_name, archive_path)
+
+    shutil.unpack_archive(archive_path, dataset_dir, format="zip")
+
+
+def _download_archive(
+    dataset_name,
+    archive_path,
+    retry=3,
+    delay=1,
+    timeout=30,
+    chunk_size=4096,
+):
+    metadata = DATASET_INFO[dataset_name]
+
+    for idx in range(1, retry + 1):
+        for target_url in metadata["urls"]:
+            print(
+                f"Downloading {dataset_name!r} from {target_url} (retry {idx}/{retry})"
+            )
+            try:
+                error_flag = False
+                _stream_download(archive_path, target_url, timeout, chunk_size)
```

Comment:
maybe checking that the download was successful ie that the checksum is correct should be the responsibility of `stream_download`. because the temp file should only be renamed if it has the correct checksum, and should be removed otherwise. also it will simplify a bit this function as we would not need the "error flag" any more and just return if `_stream_download` succeeds